src/packidx.rs

```diff
@@ -439,6 +442,32 @@ impl PackIndex {
         }
         Ok(None)
     }
+    /// Computes the checksum of the contents of the snapshot.
+    pub fn compute_snapshot_checksum(&self, snapshot: &str) -> Option<ObjectChecksum> {
+        // Get the FIleHandles
+        let handles = self.resolve_snapshot(snapshot)?;
+        // Map FileHandle to FileEntry, which contains path and checksum
+        let mut entries = self.entries_from_handles(handles.iter()).ok()?;
+        entries.sort_by(|a, b| a.checksum.cmp(&b.checksum).then(a.path.cmp(&b.path)));
+        let mut hasher = entries.into_iter().fold(Sha1::new(), |mut hasher, entry| {
+            hasher.input(os_str_as_bytes(&entry.path).unwrap_or(&[]));
+            hasher.input(&entry.checksum);
+            hasher
+        });
+        let mut checksum: ObjectChecksum = [0; 20];
+        hasher.result(&mut checksum);
+        Some(checksum)
+    }
+}
+
+#[cfg(unix)]
+fn os_str_as_bytes(os_str: &OsStr) -> Option<&[u8]> {
+    Some(std::os::unix::ffi::OsStrExt::as_bytes(os_str))
+}
+
+#[cfg(not(unix))]
+fn os_str_as_bytes(os_str: &OsStr) -> Option<&[u8]> {
+    None()
```

Comment:
I think this warrants a comment at least. I read that the effect of it is that on windows, two snapshots with the same name, with an identical set of checksums in it, but different path names, would be declared equivalent where they are not.

My preference would be that we don't use OsStr in our internal paths, but clearly that's a harder fix. So I'm willing to live with this minor gotcha as-is for now.

Reply:
I "fixed" this by using to_string_lossy, which may allocate but is at least more correct.